lib/datastore-backup.js

```diff
@@ -3,30 +3,34 @@
 const _ = require('lodash');
 const child_process = require('child_process');
 require('colors');
+// Imports the Google Cloud client library
+// @see https://googleapis.dev/nodejs/datastore/latest/
+const {Datastore} = require('@google-cloud/datastore');
 
 /**
- * create a backup of kinds from project into bucket
+ * create a backup of kinds from project into bucket; depends only on Datastore API client, rather
+ * than calling anything on cmd line - so could be run in NodeJS env that can't call out to OS
+ * (eg, Cloud Functions or something)
  *
  * @param kinds
  * @param project
  * @param bucket
  * @param options
  * @returns {string|*}
  */
-const backup = (kinds, project, bucket, options) => {
+const backup = async (kinds, project, bucket, options) => {
 
-  let authOptions = ' --project ' + project;
-  if (!_.isUndefined(options.account)) {
-    authOptions += '--account ' + options.account;
-  }
-
-  let command = 'gcloud datastore export ' + authOptions + ' --kinds="' + kinds.join(',') + '" --async gs://' + bucket;
+  const datastore = new Datastore({projectId : project});
 
-  if (options.debug) {
-    console.log(command.yellow);
-    return '';
-  } else {
-    return child_process.execSync(command).toString('utf8');
+  try {
+    return datastore.export({
```

Comment:
Cool docs that force you to read code samples

<img width="506" alt="Screenshot 2021-05-13 at 14 48 49" src="https://user-images.githubusercontent.com/1151427/118127964-981a7a00-b3fa-11eb-8400-6f42ca8e232d.png">